core/field_variable.js

```diff
@@ -341,6 +341,16 @@ Blockly.FieldVariable.prototype.setTypes_ = function(opt_variableTypes,
   this.variableTypes = variableTypes;
 };
 
+/**
+ * A callback to update the name of a variable. Should only be called by
+ * the block.
+ * @package
+ */
+Blockly.FieldVariable.prototype.updateVariableName = function() {
+  this.text_ = this.variable_.name;
```

Comment:
Can you rename this to `refreshVariableName`?  I want to indicate that it isn't actually changing the name on the model, just re-grabbing it because the model changed out from underneath the field.  And update the comment accordingly.